### DISCUSSION TITLE
Added kerning and line spacing (line height) to XTextFormatter

### DISCUSSION THREAD

**Discussion by Jonowa** [empira/PDFsharp#163] [Show and tell]
Hi community,

I've added some nice features to the XTextFormatter class. You can find them in my fork on:
[https://github.com/Jonowa/PDFsharp](https://github.com/Jonowa/PDFsharp)

Usage:
```C#
var formatter = new XTextFormatter(gfx) {
    Kerning = 40,
    LineHeight = 12
};
```

I've already written an extension class for XGraphics for the PdfSharp V1.5. There you can pass html to the XGraphics-DrawText function. Maybe this could be a solution for adding more features to the XTextFormatter like changing font, size, colors, and many more.

Regards
Joerg

**Comment by TH-Soft** [comment]
Hi!
Do not expect any enhancements to XTextFormatter to be ever included in the PDFsharp repository.
See line 11 of the source code: "/// Represents a very simple text formatter."

I added support for line heights and height calculation (page breaks) back in 2015.
https://forum.pdfsharp.net/viewtopic.php?f=8&t=3192

For kerning, see also:
https://en.wikipedia.org/wiki/Kerning
